src/core/psxhw.cc

```diff
@@ -478,7 +477,7 @@ void PCSX::HW::write16(uint32_t add, uint32_t rawvalue) {
             SIO0_LOG("sio write16 %x, %x\n", add & 0xf, value);
             break;
         case 0x1f801050:  // rx/tx data register
-            PCSX::g_emulator->m_sio1->writeData16(value);
+            PCSX::g_emulator->m_sio1->writeData8(value); // 8-bit reg, ignore upper 16 bits
```

Comment:
Maybe it's best to cast to u8 here like in line 460 for consistency & clarity. One way or another the 2 versions would be equivalent.